File: java/RemoveDuplicatesFromSortedArray.java
// Accepted
import java.util.Arrays;
// @lc id=26 lang=java
// problem: remove_duplicates_from_sorted_array
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int lo = 0;
        for (int i = 0; i<nums.length; i++) {
            if (nums[i] != nums[lo]) {
                nums[ ++lo] = nums[i];
            }
        }
        return lo + 1;
    }
}
public class RemoveDuplicatesFromSortedArray {
    public static void main(String[] argv) {
        int [] nums = {1, 2, 3, 3, 3, 4};
        new Solution().removeDuplicates(nums);
        System.out.println(Arrays.toString(nums));
    }
}

File: go/src/remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.go
// 2017-11-14 18:54
package main

import "fmt"

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	start := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != nums[start] {
			start++
			nums[i], nums[start] = nums[start], nums[i]
		}
	}
	fmt.Println(nums)
	return start + 1
}
func main() {
	fmt.Println(removeDuplicates([]int{1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 5}))
}
